### PR TITLE
Refactor wait logic for updating installations

### PR DESCRIPTION
**Pull request by gabrieljackson**
This uses a new provisioner endpoint to more efficiently determine
if it is safe for the fleet controller to make further installation
changes. The change also includes a few bug fixes resulting from
cloud provisioner changes.

Fixes https://mattermost.atlassian.net/browse/MM-36450

### CODE REVIEW

File: model/helpers.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+func InstallationsUpdatingIsBelowMax(max int64, client *cmodel.Client, logger log.FieldLogger) bool {
+	status, err := client.GetInstallationsStatus()
+	if err != nil {
+		logger.WithError(err).Error("Failed to get updating installation count")
+		return false
+	}
+
+	logger.Debugf("%d installations are currently updating (max %d)", status.InstallationsUpdating, max)
+
+	if status.InstallationsUpdating >= max {
```
Comment (gigawhitlocks) [inline model/helpers.go:19]: You could just return here, `return status.InstallationsUpdating < max`

**Review by gigawhitlocks — APPROVED** [review]
One very small suggestion :)

LGTM though

File: model/helpers.go (hunk 1 above)
Comment (gabrieljackson) [reply model/helpers.go:19]: Man, I always forget to do this. Thanks for the reminder!

**Review by stylianosrigas — APPROVED** [review]
Nice cleanup, LGTM 👌

**Pull request merged**